src/XMakeBuildEngine/UnitTestsPublicOM/Definition/ProjectItem_Tests.cs

```diff
@@ -442,42 +441,98 @@ public void ExcludeVectorWithIncludeVector()
             ObjectModelHelpers.AssertItems(new[] { "a", "b", "c", "z", "a", "c", "u" }, items);
         }
 
-        /// <summary>
-        /// Include and Exclude containing wildcards
-        /// </summary>
-        [Fact]
-        public void Wildcards()
-        {
-            string directory = null;
-            string file1 = null;
-            string file2 = null;
-            string file3 = null;
-
+        private const string ExcludeWithWildCards = @"
+                    <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003' >
+                        <ItemGroup>
+                            <i Include='{0}' Exclude='{1}'/>
+                        </ItemGroup>
+                    </Project>
+                ";
+        [Theory]
+        [InlineData(
+            ExcludeWithWildCards,
+            "a.*",
+            "*.1",
+            new[] { "a.1", "a.2", "a.1" },
+            new[] { "a.2" })]
+        [InlineData(
+            ExcludeWithWildCards,
+            @"**\*.cs",
+            @"a\**",
+            new[] { "1.cs", @"a\2.cs", @"a\b\3.cs", @"a\b\c\4.cs" },
```

Comment:
This with all of its embedded slashes will be a merge nightmare into xplat. I think you'll have to go all the way to the enumerable-driven theory. See [https://github.com/Microsoft/msbuild/blob/63c0ab3ef9ca42d4c6d99cccc7dbbd6e5ed26df3/src/XMakeBuildEngine/UnitTests/ExpressionTreeExpression_Tests.cs](https://github.com/Microsoft/msbuild/blob/63c0ab3ef9ca42d4c6d99cccc7dbbd6e5ed26df3/src/XMakeBuildEngine/UnitTests/ExpressionTreeExpression_Tests.cs) for an example.